src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -64,5 +64,40 @@ public DriverAvailability postDriverAvailability(
         DriverAvailability savedDriverAvailability = driverAvailabilityRepository.save(driverAvailability);
 
         return savedDriverAvailability;
+        }
+
+    @Operation(summary= "Delete Driver Availability")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @DeleteMapping("")
+    public Object deletemenuitem(
+        @Parameter(name="id") @RequestParam Long id) 
+        {
+        DriverAvailability drivav = driverAvailabilityRepository.findById(id)
+            .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
+        
+         driverAvailabilityRepository.delete(drivav);
+        return genericMessage("Driver Availability with id %s deleted".formatted(id));
+        }
+    
+
+    @Operation(summary= "Update Driver Availability")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @PutMapping("")
+    public DriverAvailability updatemenuitem(
```

Comment:
Drivers should only be able to update their own, admins should be able to edit anything

Reply:
Same here as delete